spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/NNPrimitive.scala

```diff
@@ -1006,9 +1006,9 @@ private[nn] object NNPrimitive {
 
       var y = 0
       while (y < frameSize) {
-        val maxIndex = xp + y
+        val maxIndex = indices(xp + y).toInt - 1
```

Comment:
why changing this? do we need to add unitest to cover it?

Reply:
We have unit test, this just a bug, I fixed it, keep float same as double

Reply:
ok. just wondering why the existing unittest didn't capture this bug.